edd-additional-shortcodes.php

```diff
@@ -7,104 +7,102 @@
+	public string $plugin_dir;
+	public string $plugin_url;
+	public string $version = '1.4.1';
 
-				self::$instance->properties();
-				self::$instance->includes();
-				self::$instance->shortcodes = new EDD_Additional_Shortcodes_Core();
+	public EDD_Additional_Shortcodes_Core $shortcodes;
+	public array $integrations = [];
 
-				if ( class_exists( 'EDD_Software_Licensing' ) ) {
-					self::$instance->integrations['software_licensing'] = new EDD_Additional_Shortcodes_SL();
-				}
-			}
-
-			return self::$instance;
+	/**
+	 * Private constructor to enforce singleton.
+	 */
+	private function __construct() {
+		if ( ! defined( 'EDD_VERSION' ) ) {
+			return;
 		}
-
-		/**
-		 * Setup properties for the class
-		 *
-		 * @since 1.4
-		 */
-		private function properties() {
-			self::$instance->version = '1.4.1';
-			self::$instance->plugin_dir = trailingslashit( plugin_dir_path( __FILE__ ) );
-			self::$instance->plugin_url = trailingslashit( plugin_dir_url( __FILE__ ) );
+	}
+
+	/**
+	 * Get active instance
+	 *
+	 * @since 1.0
+	 */
+	public static function instance(): ?EDD_Additional_Shortcodes {
+		if ( null === self::$instance ) {
+			self::$instance = new self();
+			self::$instance->setup_properties();
+			self::$instance->includes();
+			self::$instance->init();
 		}
-
-		/**
-		 * Include the necessary files
-		 *
-		 * @since 1.4
-		 */
-		private function includes() {
-			include_once self::$instance->plugin_dir . 'includes/shortcodes.php';
-
-			include_once self::$instance->plugin_dir . 'includes/backwards-compatibility.php';
-
-			if ( class_exists( 'EDD_Software_Licensing' ) ) {
-				include_once self::$instance->plugin_dir . 'includes/integrations/software-licensing.php';
-			}
+		return self::$instance;
+	}
+
+	/**
+	 * Setup properties for the class
+	 *
+	 * @since 1.4
+	 */
+	private function setup_properties(): void {
+		$this->plugin_dir = trailingslashit( plugin_dir_path( __FILE__ ) );
+		$this->plugin_url = trailingslashit( plugin_dir_url( __FILE__ ) );
+	}
+
+	/**
+	 * Include the necessary files
+	 *
+	 * @since 1.4
+	 */
+	private function includes(): void {
+		require_once $this->plugin_dir . 'includes/shortcodes.php';
+		require_once $this->plugin_dir . 'includes/backwards-compatibility.php';
+
+		if ( class_exists( 'EDD_Software_Licensing' ) ) {
+			require_once $this->plugin_dir . 'includes/integrations/software-licensing.php';
 		}
+	}
 
-		/**
-		 * Maybe execute the shortcode as a shortcode, or simply return the content.
-		 *
-		 * @since 1.4
-		 * @return string
-		 */
-		public function maybe_do_shortcode( $content ) {
-			$do_shortcode = apply_filters( 'edd_asc_do_shortcode', true );
-
-			if ( $do_shortcode ) {
-				return do_shortcode( $content );
-			}
+	/**
+	 * Initialize shortcodes and integrations
+	 */
+	private function init(): void {
+		$this->shortcodes = new EDD_Additional_Shortcodes_Core();
 
-			return $content;
+		if ( class_exists( 'EDD_Software_Licensing' ) ) {
```

Comment:
What: Change the return type hinting of `may_do_shortcode` to enforce stricter type checking on the input.

Why: The function accepts a string but the type hint is inferred. This may allow unintended types causing potential issues later on.

How: Update the method declaration from `public function maybe_do_shortcode($content)` to `public function maybe_do_shortcode(string $content): string`.